ui/src/store/ConversationMessageStore.ts

```diff
@@ -211,42 +198,25 @@ export function createConversationMessageStore(
       agents: agentPubKeys,
     });
 
-    const deletionStatus = await client.getDeleteStatus(
-      cellId,
-      decodeHashFromBase64(actionHashB64),
-    );
-
     messages.update((m) => ({
       ...m,
-      [key1]: {
-        ...m[key1],
-        [actionHashB64]: {
-          ...m[key1]?.[actionHashB64],
-          deletedAt: deletionStatus?.hashed.content.timestamp,
-        },
-      },
+      [key1]: Object.fromEntries(
+        Object.entries(m[key1]).filter(([key2]) => key2 !== actionHashB64),
+      ),
     }));
   }
 
-  async function handleMessageDeletedSignalReceived(
-    key1: CellIdB64,
-    actionHashB64: ActionHashB64,
-    deletionTimestamp: number,
-  ) {
+  async function handleMessageDeletedSignalReceived(key1: CellIdB64, actionHashB64: ActionHashB64) {
     const currentMessages = get(messages).data[key1];
     if (currentMessages[actionHashB64] === undefined) {
       return;
     }
 
     messages.update((m) => ({
       ...m,
-      [key1]: {
-        ...m[key1],
-        [actionHashB64]: {
-          ...m[key1]?.[actionHashB64],
-          deletedAt: deletionTimestamp,
-        },
-      },
+      [key1]: Object.fromEntries(
+        Object.entries(m[key1]).filter(([key2]) => key2 !== actionHashB64),
+      ),
     }));
```

Comment:
Wouldn't below will be more memory efficient? If I am not wrong, current impl has a time complexity of O(n), as it loops through all entries before reconstructing the object. 

Below impl will reduce it to O(1).
```suggestion
      messages.update((m) => {
      const k = { ...m[key1] };
      delete k[actionHashB64];
      return {
        ...m,
        [key1]: k,
      };
    });
```

Reply:
Yes good catch -- I actually made that change debugging something else but it wasn't necessary. Using `delete` isn't reactive inside svelte components, but its fine inside the svelte/store update fn. I just pushed the change.